src/PactNet.Native/Scenarios.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using PactNet.Models;
+
+namespace PactNet.Native
+{
+    /// <summary>
+    /// Defines the scenarios static list for messaging support
+    /// </summary>
+    public class Scenarios
+    {
+        /// <summary>
+        /// The available scenarios
+        /// </summary>
+        internal readonly List<Scenario> AllScenarios = new List<Scenario>();
+
+        /// <summary>
+        /// Add a scenario
+        /// </summary>
+        /// <param name="scenario">the scenario to add</param>
+        public void AddScenario(Scenario scenario)
+        {
+            if (scenario == null)
+            {
+                throw new ArgumentNullException(nameof(scenario));
+            }
+
+            if (Exist(scenario.Description))
+            {
+                throw new InvalidOperationException($"Scenario \"{scenario.Description}\" already added");
+            }
+
+            AllScenarios.Add(scenario);
+        }
+
+        /// <summary>
+        /// Add multiple scenarios
+        /// </summary>
+        /// <param name="scenarios">the scenario list to add</param>
+        public void AddScenarios(IReadOnlyCollection<Scenario> scenarios)
+        {
+            if (scenarios == null || scenarios.Any() == false)
+            {
+                throw new ArgumentException("scenarios cannot be null or empty");
+            }
+
+            if (scenarios.Any(x => Exist(x.Description)))
+            {
+                throw new InvalidOperationException("A scenario has already been added");
+            }
+
+            AllScenarios.AddRange(scenarios);
+        }
+
+        /// <summary>
+        /// Invokes the scenarios
+        /// </summary>
+        /// <param name="description">the name of the scenario</param>
+        /// <returns>a dynamic message object</returns>
+        public dynamic InvokeScenario(string description)
+        {
+            if (string.IsNullOrWhiteSpace(description))
+            {
+                throw new ArgumentNullException(nameof(description));
+            }
+
+            if (!Exist(description))
+            {
+                throw new InvalidOperationException($"Scenario \"{description}\" not found. You need to add the scenario first");
+            }
+
+            return AllScenarios.Single(x => x.Description == description).InvokeScenario();
```

Comment:
Instead of searching the list twice (both here and in `Exists`) wouldn't it be better to use something like `FirstOrDefault` and then throw on default?